src/lib/crud.route.factory.ts

```diff
@@ -219,6 +208,19 @@ export class CrudRouteFactory {
             logger: this.crudLogger,
         };
 
+        let paginationType;
+        if (crudMethod === Method.READ_MANY || crudMethod === Method.SEARCH) {
+            paginationType = this.crudOptions.routes?.[crudMethod]?.paginationType ?? CRUD_POLICY[crudMethod].default.paginationType;
+            const isPaginationType = (
+                <TEnum extends Record<string, unknown>>(enumType: TEnum) =>
+                (nextCursor: unknown): nextCursor is TEnum[keyof TEnum] =>
+                    Object.values(enumType).includes(nextCursor as TEnum[keyof TEnum])
+            )(PaginationType);
+            if (!isPaginationType(paginationType)) {
+                throw new TypeError(`invalid PaginationType ${paginationType}`);
+            }
+        }
+
```

Comment:
Now the logic is called for each method, so paginationType can be reflected individually.